app/frontend/wallet/helpers/mnemonicToWalletSecret.js

```diff
@@ -0,0 +1,33 @@
+const {decodePaperWalletMnemonic, mnemonicToRootKeypair} = require('cardano-crypto.js')
+
+const {isMnemonicInPaperWalletFormat} = require('../mnemonic')
+const {validateMnemonic} = require('../mnemonic')
+const derivationSchemes = require('../derivation-schemes')
+
+const guessDerivationSchemeFromMnemonic = (mnemonic) => {
+  return mnemonic.split(' ').length === 12 ? derivationSchemes.v1 : derivationSchemes.v2
+}
+
+const _mnemonicToWalletSecret = async (mnemonic, derivationScheme) => {
+  if (await isMnemonicInPaperWalletFormat(mnemonic)) {
+    mnemonic = await decodePaperWalletMnemonic(mnemonic)
+  }
+
+  return mnemonicToRootKeypair(mnemonic, derivationScheme.number)
+}
+
+const mnemonicToWalletSecret = async (mnemonic) => {
```

Comment:
I would name it `mnemonicToWalletSecretDef` to keep consistent naming with the output structure which is called `walletSecretDef` in the code